test/parallel/test-tls-multi-key.js

```diff
@@ -2,52 +2,162 @@
 const common = require('../common');
 const assert = require('assert');
 
+// Test multi-identity ('key')/multi-algorithm scenarios.
+
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
 const tls = require('tls');
 const fs = require('fs');
 
-const options = {
+// Key and cert are ordered as ec, rsa.
+test({
   key: [
     fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
   ],
   cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
     fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
-    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem')
-  ]
-};
-
-const ciphers = [];
-
-const server = tls.createServer(options, function(conn) {
-  conn.end('ok');
-}).listen(0, function() {
-  const ecdsa = tls.connect(this.address().port, {
-    ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
-    rejectUnauthorized: false
-  }, function() {
-    ciphers.push(ecdsa.getCipher());
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert are in opposite order from each other (order is not required).
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key, cert, and pfx options can be used simultaneously.
+test({
+  key: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-key.pem'),
+  ],
+  cert: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+  ],
+  pfx: fs.readFileSync(common.fixturesDir + '/keys/agent1.pfx'),
+  passphrase: 'sample',
+  ca: [
+    fs.readFileSync(common.fixturesDir + '/keys/ec-cert.pem'),
+    fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem'),
+  ],
+});
+
+// Key and cert with mixed algorithms, and cert chains with intermediate CAs
+test({
```

Comment:
@shigeki I don't understand your comment "so this cannot formed right cert chain." Can you elaborate on what is not right?

I don't know if you intentionally said "chain" (singular), because there are two, completely independent cert chains there, the chains you draw out above. They do not share an intermediate cert or a root cert, deliberately, this is exactly the scenario this test is intended to test: two completely independent cert chains. What is not "right"? Also, the tests pass... so the certs can't be that wrong. :-)

I don't understand your suggested change. `.rootca` is not a supported option by node, but perhaps your concern is that the .ca property was being passed to (but ignored) by the server, I removed that to make it more clear that .cert+.key is  used by the server, and .ca is  used by the client. See f6d239e